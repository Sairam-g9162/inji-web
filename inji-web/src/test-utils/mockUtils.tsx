import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

const renderWithProvider = (ui: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {ui}
            </Router>
        </Provider>,
        options
    );
};

export * from '@testing-library/react';
export { renderWithProvider };
