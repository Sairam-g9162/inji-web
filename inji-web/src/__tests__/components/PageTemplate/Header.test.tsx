import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { Header } from "../../../components/PageTemplate/Header";
import { mockUseNavigate, wrapUnderRouter } from '../../../test-utils/mockUtils';
import { renderWithProvider } from '../../../test-utils/mockUtils';

const mockedUsedNavigate = jest.fn();

jest.mock('../../../components/Common/LanguageSelector', () => ({
    LanguageSelector: () => <></>,
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

describe("Header Container Layout Test", () => {
    test('check the presence of the three menus in the header', () => {
        const { asFragment } = renderWithProvider((<Header />));
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Test Header Container Functionality", () => {
    test("Testing length of the Header menu elements", () => {
        renderWithProvider((<Header />));
        const headerElementLi = screen.getByTestId("Header-Menu-Elements");
        expect(headerElementLi.children.length).toBe(3);
    });

    // Uncomment and fix these tests if needed
    // test('check the presence of the help', () => {
    //     render(wrapUnderRouter(<Header />));
    //     const headerElementLi = screen.getByTestId("Header-Menu-Help");
    //     expect(headerElementLi).toBeInTheDocument();
    //     expect(headerElementLi).toHaveTextContent("Header.help");
    //     const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
    //     expect(headerElementDiv).toBeInTheDocument();

    //     jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
    //     fireEvent.click(headerElementDiv);

    //     expect(mockedUsedNavigate).toHaveBeenCalled();
    //     expect(mockedUsedNavigate).toHaveBeenCalledWith("/help");
    // });

    // test('check the presence of the About Inji', () => {
    //     render(wrapUnderRouter(<Header />));
    //     const headerElementLi = screen.getByTestId("Header-Menu-AboutInji");
    //     expect(headerElementLi).toBeInTheDocument();
    //     expect(headerElementLi).toHaveTextContent("Header.aboutInji");
    //     const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
    //     expect(headerElementDiv).toBeInTheDocument();

    //     jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
    //     fireEvent.click(headerElementDiv);

    //     expect(mockedUsedNavigate).toHaveBeenCalled();
    // });

    test('check the presence of the Language Selector', () => {
        renderWithProvider((<Header />));
        const headerElementLi = screen.getByTestId("Header-Menu-LanguageSelector");
        expect(headerElementLi).toBeInTheDocument();
        const headerElementDiv = screen.getByTestId("Header-Menu-Help-div");
        expect(headerElementDiv).toBeInTheDocument();

        jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockUseNavigate);
        fireEvent.click(headerElementDiv);

        expect(mockedUsedNavigate).toHaveBeenCalled();
    });
});
