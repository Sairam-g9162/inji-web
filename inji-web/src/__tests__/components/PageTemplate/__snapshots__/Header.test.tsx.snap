// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container Layout Test check the presence of the three menus in the header 1`] = `
<DocumentFragment>
  <header>
    <div
      class="fixed top-0 left-0 right-0 bg-iw-background py-4 shadow-sm"
      data-testid="Header-Container"
    >
      <div
        class="container mx-auto flex justify-between items-center px-4"
      >
        <div
          class="flex flex-row justify-center items-center"
          data-testid="Header-InjiWeb-Logo-Container"
        >
          <div
            class="sm:hidden"
            role="button"
            tabindex="0"
          >
            <svg
              fill="currentColor"
              height="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              />
            </svg>
          </div>
          <div
            role="button"
            tabindex="0"
          >
            <img
              alt="Inji Web Logo"
              class="h-13 w-28 cursor-pointer"
              data-testid="Header-InjiWeb-Logo"
              src="InjiWebLogo.png"
            />
          </div>
        </div>
        <nav>
          <ul
            class="flex space-x-4 items-center font-semibold"
            data-testid="Header-Menu-Elements"
          >
            <li
              data-testid="Header-Menu-AboutInji"
            >
              <a
                class="text-iw-title hidden sm:inline-block"
                href="https://docs.mosip.io/inji/inji-web/overview"
                rel="noreferrer"
                target="_blank"
              >
                About Inji Web
              </a>
            </li>
            <li
              data-testid="Header-Menu-Help"
            >
              <div
                class="text-iw-title cursor-pointer hidden sm:inline-block"
                data-testid="Header-Menu-Help-div"
                role="button"
                tabindex="0"
              >
                Help
              </div>
            </li>
            <li
              data-testid="Header-Menu-LanguageSelector"
            />
          </ul>
        </nav>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
